src/python/pants/backend/scala/compile/scalac_plugins.py

```diff
@@ -46,6 +47,8 @@ async def parse_global_scalac_plugins(scalac_plugins: Scalac) -> _LoadedGlobalSc
         if target.has_field(ScalacPluginArtifactField):
             artifact_address_inputs.append(cast(str, target[ScalacPluginArtifactField].value))
             names.append(target.get(ScalacPluginNameField).value or target.address.target_name)
+        elif target.has_field(JvmArtifactArtifactField):
```

Comment:
I added this so that the `jvm_artifact` for scala-library could be provided for scalac plugins. Not sure this is the right thing to do, but it is necessary for this PR to pass tests as it is currently written.

Reply:
cc @stuhood

Reply:
That makes sense to me. Can you get the tests to pass with removing this code?

Reply:
Will see. I just need to exclude scalac_plugin from getting injected with a scala-library dependency.

Reply:
As it turns out, the issue with scalac plugins was with the _tests_. They did not cleanly separate the default resolve's lockfile from the scalac plugin resolve. Having done that now, tests for scalac plugins pass without any need for the scalac plugin resolve needing a scala-library requirement.